torch/_inductor/remote_cache.py

```diff
@@ -0,0 +1,46 @@
+import os
+from abc import abstractmethod
+
+
+class RemoteCacheBackend:
+    """
+    A backend implementation for accessing a remote/distributed cache.
+    """
+
+    def __init__(self, cache_id: str):
+        pass
+
+    @abstractmethod
+    def get(self, key: str):
+        pass
+
+    @abstractmethod
+    def put(self, key: str, data: bytes):
+        pass
+
+
+class RedisRemoteCacheBackend(RemoteCacheBackend):
+    """
+    A Redis implementation of a remote/distributed cache.
+    """
+
+    def __init__(self, cache_id: str):
+        import redis
+
+        self._cache_id = cache_id
+        self._key_fmt = os.environ.get(
+            "TORCHINDUCTOR_REDIS_KEY_FORMAT", "pt2:{cache_id}:{key}"
+        )
+        self._redis = redis.Redis(
+            host=os.environ.get("TRITON_REDIS_HOST", "localhost"),
```

Comment:
Hmmm @oulgen, should we have changed these to INDUCTOR_REDIS_HOST/PORT or PT2_REDIS_HOST/PORT? "triton" is kinda confusing -- except that we almost certainly would want the same location as the remote triton cache...

Perhaps we could accept either env var? That is: TRITON_REDIS_HOST if available, else INDUCTOR_REDIS_HOST if available, else localhost?